tests/python_client/testcases/test_index.py

```diff
@@ -1455,6 +1506,65 @@ def test_create_inverted_index_no_all_vector_index(self, scalar_index):
                                                    f"[{vector_name_list[0]} {vector_name_list[1]}], "
                                                    f"please create index firstly"})
 
+    @pytest.mark.tags(CaseLabel.L1)
+    @pytest.mark.xfail(reason="issue 1931#issuecomment-1960816453")
+    def test_set_non_exist_index_mmap(self):
+        """
+        target: enabling mmap for non-existent indexes
+        method: enabling mmap for non-existent indexes
+        expected: raise exception
+        """
+        self._connect()
+        c_name = cf.gen_unique_str(prefix)
+        collection_w, _ = self.collection_wrap.init_collection(c_name, schema=default_schema)
+        collection_w.insert(cf.gen_default_list_data())
+        collection_w.create_index(ct.default_float_vec_field_name, index_params=ct.default_flat_index,
+                                  index_name=ct.default_index_name)
+        collection_w.alter_index("random_index_345", {'mmap.enabled': True})
+
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_load_mmap_index(self):
+        """
+        target: after loading, enable mmap for the index
+        method: 1. data preparation and create index
+        2. load collection
+        3. enable mmap on index
+        expected: raise exception
+        """
+        c_name = cf.gen_unique_str(prefix)
+        collection_w = self.init_collection_wrap(c_name, schema=default_schema)
+        collection_w.insert(cf.gen_default_list_data())
+        collection_w.create_index(ct.default_float_vec_field_name, index_params=ct.default_flat_index,
+                                  index_name=ct.default_index_name)
+        collection_w.load()
+        collection_w.alter_index(binary_field_name, {'mmap.enabled': True},
+                                 check_task=CheckTasks.err_res,
+                                 check_items={ct.err_code: 104,
+                                              ct.err_msg: f"can't alter index on loaded collection"})
+
+    @pytest.mark.tags(CaseLabel.L2)
+    def test_turning_on_mmap_for_scalar_index(self):
+        """
+        target: turn on mmap for scalar indexes
+        method: turn on mmap for scalar indexes
+        expected: raise exception
+        """
+        collection_w = self.init_collection_general(prefix, is_index=False, is_all_data_type=True)[0]
+        scalar_index_params = {"index_type": "INVERTED"}
```

Comment:
how about Trie index.

Reply:
> how about Trie index.

Trie index seems to be a scalar index as well, I will add it